lib/dockerregistry/inventory.go

```diff
@@ -1561,6 +1829,57 @@ func (sc *SyncContext) FilterPromotionEdges(
 	return sc.getPromotionCandidates(edges)
 }
 
+// EdgesToRegInvImage takes the destination endpoints of all edges and converts
+// their information to a RegInvImage type. It uses only those edges that are
+// trying to promote to the given destination registry.
+func EdgesToRegInvImage(
+	edges map[PromotionEdge]interface{},
+	destRegistry string) RegInvImage {
+
+	rii := make(RegInvImage)
+
+	destRegistry = strings.TrimRight(destRegistry, "/")
```

Comment:
Don't you want to consistently add the `/`, so `gcr.io/foo` doesn't match `gcr.io/foobar` ?

Reply:
This is actually sanitizing the `*manifestBasedSnapshotOf` pointer value as we pass that on directly to this function from cip.go.